pennylane/grouping/utils.py

```diff
@@ -0,0 +1,589 @@
+            )
+        )
+
+    if not (
+        np.array_equal(pauli_vec_1, pauli_vec_1.astype(bool))
+        and np.array_equal(pauli_vec_2, pauli_vec_2.astype(bool))
+    ):
+        raise ValueError(
+            "Vectors a and b must have strictly binary components, instead got {} and {}".format(
+                pauli_vec_1, pauli_vec_2
+            )
+        )
+
+    n_qubits = int(len(pauli_vec_1) / 2)
+
+    for i in range(n_qubits):
+
+        a_x_i = pauli_vec_1[i]
+        a_z_i = pauli_vec_1[n_qubits + i]
+        b_x_i = pauli_vec_2[i]
+        b_z_i = pauli_vec_2[n_qubits + i]
+
+        if not (
+            ((a_x_i == a_z_i == 0) or (b_x_i == b_z_i == 0))
+            or ((a_x_i == b_x_i) and (a_z_i == b_z_i))
+        ):
+            return False
+
+    return True
+
+
+def convert_observables_to_binary_matrix(observables, n_qubits=None, wire_map=None):
+    """Converts a list of Pauli words to the binary vector representation and yields column matrix
+        of the binary vectors.
+
+    The dimension of the binary vectors will be implied from the highest wire being acted on
+    non-trivially by the Pauli words in observables.
+
+    **Usage example:**
+
+    >>> convert_observables_to_binary([PauliX(0) @ PauliY(2), PauliZ(0) @ PauliZ(1) @ PauliZ(2)])
+
+    array([[1., 0.],
+           [1., 0.],
+           [0., 0.],
+           [0., 1.],
+           [1., 1.],
+           [0., 1.]])
+
+    Args:
+        observables (list[Union[Identity, PauliX, PauliY, PauliZ, Tensor]]): the list of Pauli
+            words.
+
+    Keyword args:
+        n_qubits (int): number of qubits to specify dimension of binary vector representation.
+        wire_map (dict): dictionary containing all wire labels used in the Pauli words as keys, and
+                         unique integer labels as their values.
+
+
+    Returns:
+        array[array[bool]]: a matrix whose columns are Pauli words in binary vector representation.
+
+    """
+
+    m_cols = len(observables)
+
+    for c, obs in enumerate(observables):
+        if isinstance(obs, Tensor):
+            observables[c] = obs.prune()
+        if isinstance(obs, Identity):
+            observables[c] = Identity(0)
+
+    if wire_map is None:
+        all_wires = Wires.all_wires([pauli_word.wires for pauli_word in observables])
+        wire_map = {i: c for c, i in enumerate(all_wires)}
+
+    n_qubits_min = max(wire_map.values()) + 1
+    if n_qubits is None:
+        n_qubits = n_qubits_min
+    elif n_qubits < n_qubits_min:
+        raise ValueError(
+            "n_qubits must support the highest mapped wire index {},"
+            " instead got n_qubits={}.".format(n_qubits_min, n_qubits)
+        )
+
+    binary_mat = np.zeros((2 * n_qubits, m_cols))
+
+    for i in range(m_cols):
+        binary_mat[:, i] = pauli_to_binary(observables[i], n_qubits=n_qubits, wire_map=wire_map)
+
+    return binary_mat
+
+
+def get_qwc_compliment_adj_matrix(binary_observables):
```

Comment:
Will be recurring, could use search & replace for the document & files
```suggestion
def get_qwc_complement_adj_matrix(binary_observables):
```